ports/unix/main.c

```diff
@@ -369,6 +370,10 @@ static void pre_process_options(int argc, char **argv) {
                 print_help(argv);
                 exit(0);
             }
+            if (strcmp(argv[a], "--version") == 0) {
+                printf("MicroPython " MICROPY_GIT_TAG " on " MICROPY_BUILD_DATE "\n");
```

Comment:
I think this should use the existing banner macro, and also include the machine name, like this:
```c
printf(MICROPY_BANNER_NAME_AND_VERSION "; " MICROPY_BANNER_MACHINE "\n");
```

Reply:
(See related #15578)